util/testutil/storage.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"testing"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb/tsdbutil"
```

Comment:
I don't think that testutil should be depending on storage, as testutil is used by things like discovery that are completely unrelated to the tsdb.

Reply:
What do you suggest instead?

Reply:
It'd have to go somewhere else to keep the code well factored, what's the goal of this change? We have a teststorage already, or maybe in tsdbutil considering it's using a struct.

Reply:
> what's the goal of this change?

To make the `query` function used in tests usable outside of the tsdb package. That's implied by the description above:

"…because it needs to be reused in backfilling PR. #8084 "

Reply:
`teststorage` might be a good location for this. It would, however, have nothing to do with the actual `TestStorage` struct in there, but it would still have to do with testing the storage. I would be fine with that location. (And then perhaps move `queryChunks`, too, for consistency.)

`tsdbutil` sounds good, too. I was reluctant at first because the functions in question here are only used in tests. But that's true for some types in `tsdbutil`, too.

@aSquare14 you see, there are plenty of possibilities. However, you have to be aware of import cycles. The PR as is is creating an import cycle:

```
# github.com/prometheus/prometheus/tsdb/fileutil
package github.com/prometheus/prometheus/tsdb/fileutil (test)
        imports github.com/prometheus/prometheus/util/testutil
        imports github.com/prometheus/prometheus/storage
        imports github.com/prometheus/prometheus/tsdb/chunks
        imports github.com/prometheus/prometheus/tsdb/fileutil: import cycle not allowed in test
```

This particular one should be avoided by moving into `tsdbutil` or `teststorage`, but others might arise…